web-bundle/src/main/resources/com/graphhopper/maps/js/main-template.js

```diff
@@ -78,17 +78,21 @@ $(document).ready(function (e) {
     // todo: the idea was to highlight everything so if we start typing the example is overwritten. But unfortunately
     // this does not work. And not even sure this is so useful?
     showCustomModelExample();
-    cmEditor.validListener = function(valid) {
+    cmEditor.validListener = function (valid) {
         $("#custom-model-search-button").prop('disabled', !valid);
     };
-    $("#custom-model-button").click(function() {
+    ghRequest.cmEditor = cmEditor;
+    ghRequest.cmEditorActive = false;
+    $("#custom-model-button").click(function () {
         $("#custom-model-box").toggle();
+        ghRequest.cmEditorActive = !ghRequest.cmEditorActive;
```

Comment:
here we simply keep track whether or not the custom model box is opened or not. if it is we send a custom request, otherwise we ignore the custom model box and send the same request we did before.

Reply:
I really like this idea and solution! Then it is also easy to see what is changed due to the custom model.